core/chains/evm/gas/block_history_estimator.go

```diff
@@ -493,7 +487,15 @@ func (b *BlockHistoryEstimator) batchFetch(ctx context.Context, reqs []rpc.Batch
 
 		b.logger.Tracew(fmt.Sprintf("Batch fetching blocks %v thru %v", HexToInt64(reqs[i].Args[0]), HexToInt64(reqs[j-1].Args[0])))
 
-		if err := b.ethClient.BatchCallContext(ctx, reqs[i:j]); err != nil {
+		err := b.ethClient.BatchCallContext(ctx, reqs[i:j])
+		if errors.Is(err, context.DeadlineExceeded) {
+			// We ran out of time, return what we have
+			b.logger.Warnf("Batch fetching timed out; loaded %d/%d results", i, len(reqs))
+			for k := i; k < len(reqs); k++ {
+				reqs[k].Error = err
```

Comment:
I think it's fair to mark `j` through `len` errored as well, but might it be misleading to propagate the same returned err instance? Would it be more clear to set something simple for those, that doesn't have any extra wrapping suggesting an attempt was made?

Reply:
`i` is necessarily less than `j` so `i` thru `len(reqs)` will always cover `j` thru `len(reqs)` as well

Reply:
I mean that it may not be appropriate to set the exact same error instance on j through len, since they were not actually attempted here.

Reply:
Aaah I see, you could instead set the error to "not attempted" or something you mean?

Reply:
Yeah something like that. Could even still wrap the `context.DeadlineExceeded` maybe :shrug:

Reply:
Yeah, deadline exceeded is fine, but we don't check what else could be wrapping, so it might be misleading in a debugging scenario.